test/integration-new/cni/env_vars/env_test.go

```diff
@@ -0,0 +1,136 @@
+package env_vars
+
+import (
+	"regexp"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	AWS_VPC_ENI_MTU            = "AWS_VPC_ENI_MTU"
+	AWS_VPC_K8S_CNI_LOG_FILE   = "AWS_VPC_K8S_CNI_LOG_FILE"
+	AWS_VPC_K8S_CNI_VETHPREFIX = "AWS_VPC_K8S_CNI_VETHPREFIX"
+)
+
+var _ = Describe("cni env test", func() {
+
+	Context("CNI Environment Variables", func() {
+		It("Verifying that secondary ENI is created", func() {
+			nodes, err := f.K8sResourceManagers.NodeManager().GetAllNodes()
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, node := range nodes.Items {
+				instanceId := k8sUtils.GetInstanceIDFromNode(node)
+				instance, err := f.CloudServices.EC2().DescribeInstance(instanceId)
+				Expect(err).NotTo(HaveOccurred())
+
+				len := len(instance.NetworkInterfaces)
+				Expect(len).To(Equal(2))
+			}
+		})
+
+		It("Changing AWS_VPC_ENI_MTU and AWS_VPC_K8S_CNI_VETHPREFIX", func() {
+			currMTUVal := getEnvValueForKey(AWS_VPC_ENI_MTU)
+			Expect(currMTUVal).NotTo(Equal(""))
+
+			currVETHPrefix := getEnvValueForKey(AWS_VPC_K8S_CNI_VETHPREFIX)
+			Expect(currVETHPrefix).NotTo(Equal(""))
+
+			k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, DAEMONSET, NAMESPACE, DAEMONSET, map[string]string{
+				AWS_VPC_ENI_MTU:            "1300",
+				AWS_VPC_K8S_CNI_VETHPREFIX: "veth",
+			})
+
+			By("Deploying a BusyBox deployment", func() {
+				deploymentSpec := manifest.NewBusyBoxDeploymentBuilder().
+					Namespace("default").
+					Name("busybox").
+					Replicas(1).
+					NodeName(primaryNode.Name).Build()
+
+				_, err := f.K8sResourceManagers.
+					DeploymentManager().
+					CreateAndWaitTillDeploymentIsReady(deploymentSpec)
+
+				Expect(err).ToNot(HaveOccurred())
+
+				stdout, _, err := f.K8sResourceManagers.PodManager().PodExec("default", hostNetworkPod.Name, []string{"ifconfig"})
```

Comment:
I will check this

Reply:
Tried using grep, but we can pass only 1 command to the PodExec. Also, even if use grep, later we will need regex to get the MTU value. Since regex is restricted only to this test case. I think it should be okay.